### PR TITLE
feat: [CDS-82726]: Added Orgs and projects in different scope for module and all modes

### PR DESCRIPTION
**Pull request by avinashmadhwani02**
### Summary

Added Project and Org links in other modes as well. Previously, Project and Organization was available only in Admin mode. 
This can help in viewing the Projects and Orgs without changing the mode. 

#### Screenshots

Recording - 

https://github.com/harness/harness-core-ui/assets/62028553/f4166baa-80e6-43ea-bbcb-ce7671b9863c

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Review by SahilKr24 — APPROVED** [review]

File: jest.config.js [hunk 1]
```diff
@@ -44,6 +44,7 @@ module.exports = {
     '!src/**/TemplateYamlView.tsx',
     '!src/**/RouteDestinations.tsx',
     '!src/**/RouteDestinationsV2.tsx',
+    '!src/**/CommonRouteDestinations.tsx',
```
[the remark this answers is not in the document]
Comment (avinashmadhwani02) [reply jest.config.js:47]: There's no Business logic in RouteDestinations components that's why we are ignoring it. 

If we were to write tests for it, we would essentially be testing the mapping of path and component. It's better to focus on thoroughly testing the components that RouteDestinations is rendering.

File: src/modules/45-projects-orgs/components/ScopeSelector/ProjectScopeSelector/ProjectScopeSelector.tsx [hunk 2]
```diff
@@ -182,9 +183,7 @@ export const ProjectScopeSelector: React.FC<ProjectScopeSelectorProps> = ({ onPr
           variation={ButtonVariation.LINK}
           className={css.viewAllProjects}
           text={getString('projectsOrgs.viewAllProjects')}
-          onClick={() => {
-            history.push(routes.toProjects({ accountId, mode: NAV_MODE.ADMIN }))
-          }}
+          onClick={clickOnViewAllProjects}
```
Comment (samarth-sharma31) [inline src/modules/45-projects-orgs/components/ScopeSelector/ProjectScopeSelector/ProjectScopeSelector.tsx:186]: https://github.com/harness/harness-core-ui/pull/19145/files#r1411842757

File: src/modules/45-projects-orgs/components/ScopeSelector/OrgScopeSelector/OrgScopeSelector.tsx [hunk 3]
```diff
@@ -68,6 +71,11 @@ export const OrgScopeSelector = (props: OrgScopeSelectorProps): JSX.Element => {
             setPage(0)
           }}
         />
+        <Button
+          variation={ButtonVariation.LINK}
+          text={getString('projectsOrgs.viewAllOrgs')}
+          onClick={clickOnViewAllOrgs}
```
[the remark this answers is not in the document]
Comment (avinashmadhwani02) [reply src/modules/45-projects-orgs/components/ScopeSelector/OrgScopeSelector/OrgScopeSelector.tsx:77]: sure we can add. But how safe check will help?
onClick prop in button component is also optional

File: jest.config.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (avinashmadhwani02) [reply jest.config.js:47]: sure, adding assertions

[the remark this answers is not in the document]
Comment (chetanvekariya-harness) [reply jest.config.js:47]: I think mapping is equally important. You have coverage for component which will never break if I render <Abc> dummy component for let's say any valid route which is part of this component. Or am I missing something ?

**Review by chetanvekariya-harness — DISMISSED** [review]

**Review by deepesh-ui — APPROVED** [review]

**Review by vardanbansal-harness — APPROVED** [review]

**Review by jyotiarora-harness — APPROVED** [review]

**Comment by abhinavrastogi-harness** [comment]
@avinashmadhwani02 we should fix the order of all links to be `Account > Org > Projects` everywhere

File: src/modules/10-common/navigation/SideNavs/AdminSideNav/AdminSideNav.tsx [hunk 4]
```diff
@@ -39,15 +39,7 @@ const AdminSideNav: React.FC = (): React.ReactElement => {
...
-        <SideNav.Section>
-          <SideNav.Link to={routes.toOrgs()} label={getString('orgsText')} icon="nav-organization" />
-          <SideNav.Link to={routes.toProjects()} label={getString('projectsText')} icon="nav-project" />
-          <SideNav.Link
-            to={routes.toSettings({ mode: NAV_MODE.ADMIN })}
-            label={getString('common.accountSettings')}
-            icon={'setting'}
-          />
-        </SideNav.Section>
+        <SideNav.Section></SideNav.Section>
```
[the remark this answers is not in the document]
Comment (avinashmadhwani02) [reply src/modules/10-common/navigation/SideNavs/AdminSideNav/AdminSideNav.tsx]: removed empty section

**Comment by avinashmadhwani02** [comment]
> @avinashmadhwani02 we should fix the order of all links to be `Account > Org > Projects` everywhere

@abhinavrastogi-harness We can check with UX before changing the order. In all the UX mocks they are following the order we have implemented.

**Review by jyotiarora-harness — APPROVED** [review]

**Review by mayankVermaHarness — APPROVED** [review]

**Pull request merged**